homeassistant/components/sfr_box/config_flow.py

```diff
@@ -39,8 +52,48 @@ async def async_step_user(
                 await self.async_set_unique_id(system_info.mac_addr)
                 self._abort_if_unique_id_configured()
                 self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
-                return self.async_create_entry(title="SFR Box", data=user_input)
+                self._box = box
+                self._config.update(user_input)
+                return await self.async_step_choose_auth()
+
+        data_schema = self.add_suggested_values_to_schema(DATA_SCHEMA, user_input)
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_choose_auth(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        return self.async_show_menu(
+            step_id="choose_auth",
+            menu_options=["auth", "skip_auth"],
+        )
 
+    async def async_step_auth(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Check authentication."""
+        errors = {}
+        if user_input is not None:
+            try:
+                if (username := user_input.get(CONF_USERNAME)) and (
+                    password := user_input.get(CONF_PASSWORD)
+                ):
```

Comment:
It looks like there are required so `user_input[CONF_USERNAME]` ... makes sense here

Reply:
Ah yes. I had them optional previously, before I moved to menu.